src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -120,25 +121,39 @@ export default class HexagonLayer extends Layer {
   }
 
   updateRadiusAngle() {
-    const {hexagonVertices: vertices, dotRadius} = this.props;
+    let vertices = this.props.hexagonVertices;
+    let angle;
+    let radius;
 
-    const vertex0 = vertices[0];
-    const vertex3 = vertices[3];
+    if (!Array.isArray(vertices) || vertices.length !== 6) {
+      vertices = _getDefaultVertices(this.props.data, this.props.getCentroid);
+    }
 
-    // transform to space coordinates
-    const spaceCoord0 = this.projectFlat(vertex0);
-    const spaceCoord3 = this.projectFlat(vertex3);
+    if (!vertices) {
+      angle = 0;
+      radius = 10;
+    } else {
 
-    // distance between two close centroids
-    const dx = spaceCoord0[0] - spaceCoord3[0];
-    const dy = spaceCoord0[1] - spaceCoord3[1];
-    const dxy = Math.sqrt(dx * dx + dy * dy);
+      const vertex0 = vertices[0];
+      const vertex3 = vertices[3];
+
+      // transform to space coordinates
+      const spaceCoord0 = this.projectFlat(vertex0);
```

Comment:
Ideally we should have a test case for a METER_OFFSET mode HexagonLayer to make sure these JS project functions work in both cases. I can help with that.